packages/workbox-core/_private/DBWrapper.mjs

```diff
@@ -0,0 +1,351 @@
+   * and is invoked with three arguments:
+   *   1. An object mapping object store names to IDBObjectStore values.
+   *   2. A `done` function, that's used to resolve the promise when
+   *      when the transaction is done.
+   *   3. An `abort` function that can be called to abort the transaction
+   *      at any time.
+   *
+   * @param {Array<string>} storeNames An array of object store names
+   *     involved in the transaction.
+   * @param {string} type Can be `readonly` or `readwrite`.
+   * @param {function(Object, function(), function(*)):?IDBRequest} callback
+   * @return {*} The result of the transaction ran by the callback.
+   */
+  async transaction(storeNames, type, callback) {
+    await this.open();
+    const result = await new Promise((resolve, reject) => {
+      const txn = this._db.transaction(storeNames, type);
+      const done = (value) => resolve(value);
+      const abort = () => {
+        reject(new Error('The transaction was manually aborted'));
+        txn.abort();
+      };
+      txn.onerror = (evt) => reject(evt.target.error);
+      txn.onabort = (evt) => reject(evt.target.error);
+      txn.oncomplete = () => resolve();
+
+      const stores = {};
+      for (const storeName of storeNames) {
+        stores[storeName] = txn.objectStore(storeName);
+      }
+      callback(stores, done, abort);
+    });
+    return result;
+  }
+
+  /**
+   * Delegates async to a native IDBObjectStore method.
+   *
+   * @param {string} method The method name.
+   * @param {string} storeName The object store name.
+   * @param {string} type Can be `readonly` or `readwrite`.
+   * @param {...*} args The list of args to pass to the native method.
+   * @return {*} The result of the transaction.
+   */
+  async _call(method, storeName, type, ...args) {
+    await this.open();
+    const callback = (stores, done) => {
+      stores[storeName][method](...args).onsuccess = (evt) => {
+        done(evt.target.result);
+      };
+    };
+
+    return await this.transaction([storeName], type, callback);
+  }
+
+  /**
+   * The default onversionchange handler, which closes the database so other
+   * connections can open without being blocked.
+   *
+   * @param {Event} evt
+   */
+  _onversionchange(evt) {
+    this.close();
+  }
+
+  /**
+   * Closes the connection opened by `DBWrapper.open()`. Generally this method
+   * doesn't need to be called since:
+   *   1. It's usually better to keep a connection open since opening
+   *      a new connection is somewhat slow.
+   *   2. Connections are automatically closed when the reference is
+   *      garbage collected.
+   * The primary use case for needing to close a connection is when another
+   * reference (typically in another tab) needs to upgrade it and would be
+   * blocked by the current, open connection.
+   */
+  close() {
+    if (this._db) this._db.close();
+  }
+
+  /**
+   * Deletes the database.
+   */
+  async deleteDatabase() {
```

Comment:
This method looks like it's only being used in the tests, if this is the case it should probably be moved out of the source code and into `infra/testing/`.

Reply:
Yeah, I mention this is https://github.com/GoogleChrome/workbox/pull/968#issuecomment-341247613, but after thinking about it a little more, I realized when I release this as a standalone package I can separate out the `deleteDatabase` and `onsuccessAll` methods into standalone helpers.

That should cut down on the size a bit.